clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -829,6 +829,35 @@ public void commitTransaction() throws ProducerFencedException {
         producerMetrics.recordCommitTxn(time.nanoseconds() - commitStart);
     }
 
+    /**
+     * This method should only be called if there are no pending writes, i.e., only after calling {@link #flush()}.
+     * If there are any errors in sending messages to topics, these errors can be cleared by passing {@link CommitOption#CLEAR_SEND_ERRORS},
+     * allowing the transaction to be committed even in case of data loss.
+     * <p>
+     * If this method is used while there are pending sends, the send errors cannot be cleared.
+     *
+     * @param option The method option
+     * @throws IllegalStateException if no transactional.id has been configured or no transaction has been started
+     * @throws ProducerFencedException fatal error indicating another producer with the same transactional.id is active
+     * @throws org.apache.kafka.common.errors.UnsupportedVersionException fatal error indicating the broker
+     *         does not support transactions (i.e. if its version is lower than 0.11.0.0)
+     * @throws org.apache.kafka.common.errors.AuthorizationException fatal error indicating that the configured
+     *         transactional.id is not authorized. See the exception for more details
+     * @throws org.apache.kafka.common.errors.InvalidProducerEpochException if the producer has attempted to produce with an old epoch
+     *         to the partition leader. See the exception for more details
+     * @throws KafkaException if the producer has encountered a previous fatal or abortable error, or for any
+     *         other unexpected error
+     * @throws TimeoutException if the time taken for committing the transaction has surpassed <code>max.block.ms</code>.
+     * @throws InterruptException if the thread is interrupted while blocked
+     */
+    public void commitTransaction(CommitOption option) throws ProducerFencedException {
+        throwIfNoTransactionManager();
+        if (!accumulator.hasIncomplete() && option == CommitOption.CLEAR_SEND_ERRORS) {
```

Comment:
Is it possible for a request to be added after we call this? I guess it is ok if we don't clear those errors.